src/components/ShareMyLocationButton.js

```diff
@@ -28,20 +28,33 @@ const ShareMyLocationButton = ({ disabled = false }) => {
 
   const handleShareLiveLocation = () => {
     if (navigator.geolocation) {
-      if (trackingId === null) {
-        const newTrackingId = navigator.geolocation.watchPosition(
-          handleSuccess,
-          handleError,
-        );
-        setTrackingId(newTrackingId);
-      }
+      const trackingId = navigator.geolocation.watchPosition(
```

Comment:
Good view! if you share your location, you disable this button. It wasn't necessary to check if we already have shared our location. Thanks!